### ISSUE TITLE
[incomplete] October | 10/2015; List of Upstream Illumos changes

### ISSUE THREAD

**Issue by kernelOfTruth** [zfsonlinux/zfs#3982]
[applied]
<del> https://www.illumos.org/issues/6386 Fix function call with uninitialized value in vdev_inuse </del>

<del> https://github.com/zfsonlinux/zfs/commit/485c581c41c4da15a17f045605ce5a7562b3b8a2 </del>

[applied]
<del> https://www.illumos.org/issues/6390 Free props in ztest_init() </del>

<del> https://github.com/zfsonlinux/zfs/commit/85802aa42bf8f878022ec73ddde6bf3702f67d01 </del>

[applied ?]
<del> https://www.illumos.org/issues/5219 l2arc_write_buffers() may write beyond target_sz </del>

<del> https://github.com/illumos/illumos-gate/commit/d7d9a6d919f92d74ea0510a53f8441396048e800 </del>

[applied]
<del> https://www.illumos.org/issues/6385 Fix unlocking order in zfs_zget </del>

<del> https://github.com/zfsonlinux/zfs/commit/f3ad9cd67ae57760a7ec9e8cdb0e33aa8bbea4d0 </del>

[ZoL]
<del> https://www.illumos.org/issues/5984 zfs clone should not mount the clone if canmount == noauto
https://github.com/zfsonlinux/zfs/issues/2241
https://github.com/zfsonlinux/zfs/commit/480f62655d2a4c8fa9e3cf12f1245b455edac349 </del>

-----------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------

[open]

https://www.illumos.org/issues/6434 sa_find_sizes() may compute wrong SA header size

https://github.com/illumos/illumos-gate/commit/3502ed6e7cb3f3d2e781960ab8fe465fdc884834

(see: https://github.com/zfsonlinux/zfs/issues/1890 ; https://github.com/zfsonlinux/zfs/commit/472e7c60853af099fbdf9d52162fd39818884f4f sa_find_sizes() may compute wrong SA header size)

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6334 Cannot unlink files when over quota

https://github.com/illumos/illumos-gate/commit/6575bca01367958c7237253d88e5fa9ef0b1650a
(depends on may_delete_now)

https://www.illumos.org/issues/6298 zfs_create_008_neg and zpool_create_023_neg need to be updated for large block support

https://github.com/illumos/illumos-gate/commit/e9316f7696401f3e5e263a5939031cb8d5641a88

https://www.illumos.org/issues/5963 rmis should be removed from arcstat manpage

https://github.com/illumos/illumos-gate/commit/f34d737f84313e6f54728b1be0483c6234404577

https://www.illumos.org/issues/2605 want to resume interrupted zfs send

https://github.com/illumos/illumos-gate/commit/9c3fd1216fa7fb02cfbc78a2518a686d54b48ab8

6286 ZFS internal error when set large block on bootfs

https://github.com/illumos/illumos-gate/commit/6de9bb5603e65b16816b7ab29e39bac820e2da2b

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6281 prefetching should apply to 1MB reads

https://github.com/illumos/illumos-gate/commit/632802744ef6d17e06d6980a95f631615c3b060f

(depends on Illumos 5987 zfs prefetch code needs work , see: https://github.com/zfsonlinux/zfs/pull/3765 , https://github.com/zfsonlinux/zfs/pull/3611 , https://github.com/zfsonlinux/zfs/pull/3574)

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6280 libzfs: unshare_one() could fail with EZFS_SHARENFSFAILED

https://github.com/illumos/illumos-gate/commit/d1672efb6feac57c42788e27f739dfa3c4f3baf7

https://www.illumos.org/issues/6249 6249 async_destroy_001_pos fails because it doesn't see 'freeing' prop populated

https://github.com/illumos/illumos-gate/commit/db2417522bcef7cf091649ee369330ecefbaf183

https://www.illumos.org/issues/6248 6248 zpool_create_008_pos and zpool_create_009_neg can fail intermittently

https://github.com/illumos/illumos-gate/commit/999dc7aa56a6d16b1b0b88d246545be69aae9a4b

-----------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------

[processing]

https://www.illumos.org/issues/6292 exporting a pool while an async destroy is running can leave entries in the deferred tree

https://github.com/illumos/illumos-gate/commit/a443cc80c742af740aa82130db840f02b4389365

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6319 assertion failed in zio_ddt_write: bp->blk_birth == txg

https://github.com/illumos/illumos-gate/commit/b39b744be78c6327db43c1f69d11c2f5909f73cb

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6288 dmu_buf_will_dirty could be faster

https://github.com/illumos/illumos-gate/commit/0f2e7d03b8f588387cb8dd8dd500cbe5ff4484e0

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6295 metaslab_condense's dbgmsg should include vdev id

https://github.com/illumos/illumos-gate/commit/daec38ecb4fb5e73e4ca9e99be84f6b8c50c02fa

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6367 spa_config_tryenter incorrectly handles the multiple-lock case

https://github.com/illumos/illumos-gate/commit/e495b6e6735b803e422025a630352ef9bba788c5

-----------------------------------------------------------------------------------------

https://www.illumos.org/issues/6268 zfs diff confused by moving a file to another directory

https://github.com/illumos/illumos-gate/commit/aab04418a72c0a29040a5da7eec08efe19dbef04

-----------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------

[non-applicable (Illumos- or *BSD-specific]

<del> https://www.illumos.org/issues/6290 zdb -h overflows stack</del>

<del> https://github.com/illumos/illumos-gate/commit/3f84190c280ddf5917fb4682f34fdaa328ee195c </del>

(see: [ZOL] https://github.com/zfsonlinux/zfs/commit/017da6f063861a1dc3eea5a41c3de8b879d579bd Fix for recent zdb -h | -i crashes (seg fault) )

<del> https://www.illumos.org/issues/6315 taskq kstat uses the wrong label </del>

<del> https://github.com/illumos/illumos-gate/commit/7c5f01c17875fc96e2b6119c01a189afd6081e87 </del>

**Comment by behlendorf** [comment]
@kernelOfTruth awesome, this is a great start.  We should decide on the best way to track this kind of work.  What do you think about either enabling the wiki and tracking this on a wiki page, or perhaps creating a new upstream milestone and opening an issue per-upstream-patch?

**Comment by kernelOfTruth** [comment]
@behlendorf I'd say, we could go with a mix of both: 

A wiki where everyone ((future) devs, community, ...) can contribute to keep the per-month list of upstream changes up-to-date and opening an issue per patch

Not sure how to proceed with the milestone targets (leave it at the current approach), in my opinion however an additional voting system would be great to gauge how big the demand is for a feature or patch, also to involve the community some more - I'm uncertain whether github supports this but it would be a nice addition.

**Comment by behlendorf** [comment]
> A wiki where everyone ((future) devs, community, ...) can contribute

Sounds good.  I've enabled the wiki and created https://github.com/zfsonlinux/zfs/wiki/Illumos-Tracking as a starting point.

As for the milestones at the moment I make a best guess as to what can realistically be accomplished in a given milestone given the available resources.  But I think that's probably a separate issue than tracking the Illumos changes.

**Comment by gmelikov** [comment]
Maybe automatically open an issue on each openZFS commit with label `openZFS tracking` (for example)?

New issue - we react - easy filter - close if ported\already merged\doesn't need to merge.

Additional plus - we can discuss each openZFS commit in issue, and after that someone (who took the issue - assigned) can start porting.

There's only one minus - more issues. We can use separate repository for it.

I can make a light bot, which will track openZFS commits and open issues.

**Comment by behlendorf** [comment]
@kernelOfTruth @gmelikov I took the opportunity to throw together a little script which dynamically generates a simple [OpenZFS Commit Tracking](http://build.zfsonlinux.org/openzfs-tracking.html) page.  It won't be winning any design awards but it refreshes hourly and as long as we always add `OpenZFS XXX` to the git commit and new PR summaries it'll automatically link the commits and PRs.

My hope is this'll help us more easily track the outstanding porting work without resorting to opening a new issue for every OpenZFS commit.  Although, that's also a pretty reasonable option.  I've updated the wiki sidebar to reference the new page.
